ophyd/controls/ophydobj.py

```diff
@@ -319,6 +319,17 @@ def check_value(self, value, **kwargs):
         '''
         pass
 
+    @property
+    def state(self):
+        return {}
+
+    def configure(self, state=None):
+        # does nothing; subclasses can override if configuration is possible
+        return self.state, self.state
+
+    def deconfigure(self):
+        return self.state
+
```

Comment:
The model in bluesky is that if an object can be read, then it also needs
{trigger, describe, configure, deconfigure, state}.

On Thu, Oct 1, 2015 at 10:16 AM dchabot <notifications@github.com> wrote:

> In ophyd/controls/ophydobj.py
> <https://github.com/NSLS-II/ophyd/pull/149#discussion_r40917571>:
>
> > @@ -319,6 +319,17 @@ def check_value(self, value, **kwargs):
> >          '''
> >          pass
> >
> > +    @property
> > +    def state(self):
> > +        return {}
> > +
> > +    def configure(self, state=None):
> > +        # does nothing; subclasses can override if configuration is possible
> > +        return self.state, self.state
> > +
> > +    def deconfigure(self):
> > +        return self.state
> > +
>
> It seems meaningless (to me) to offer configure/deconfigure methods on
> objects that represent a subscription registry (OphydObject), or on
> objects that represent a single process variable ('Signal`).
>
> I realize configure/deconfigure could simply be a pass-through on those
> classes, but this just feels wrong.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NSLS-II/ophyd/pull/149/files#r40917571>.
>